### PR TITLE
DOM manipulation and data fetching

### PR DESCRIPTION
**Pull request by mohammadkazemy**
Closes: #79

### CODE REVIEW

File: mohammad/JS/JS-dataFetchingAndDOM/search.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  );
+
+  const title = document.createElement("p");
+  title.textContent = myTitle;
+
+  const body = document.createElement("p");
+  title.textContent = myBody;
+
+  const author = document.createElement("p");
+  title.textContent = myAuthor;
```
Comment (Mr-MRF-Dev) [inline mohammad/JS/JS-dataFetchingAndDOM/search.js:19]: title or author???

File: mohammad/JS/JS-dataFetchingAndDOM/search.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+  padding:2rem; margin:2rem;
+  border-radius:5rem;
+  `
+  );
+
+  const title = document.createElement("p");
+  title.textContent = myTitle;
+
+  const body = document.createElement("p");
+  title.textContent = myBody;
```
Comment (Mr-MRF-Dev) [inline mohammad/JS/JS-dataFetchingAndDOM/search.js:16]: title or body???

File: mohammad/JS/JS-dataFetchingAndDOM/search.js [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  });
+  return result["name"];
+}
+
+async function searchByAuthor(query) {
+  const result = await fetch(`${url}/users?name=${query}`).then((response) => {
+    return response.json();
+  });
+  if (response["length"] != 0) {
+    const authorID = response[0]["id"];
```
Comment (Mr-MRF-Dev) [inline mohammad/JS/JS-dataFetchingAndDOM/search.js:41]: Where is `response`?
Replace with `result`.

File: mohammad/JS/JS-dataFetchingAndDOM/search.js [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    searchByAuthor(query),
+    searchByTitle(query),
+    searchByBody(query),
+  ]).then((result) => {
+    result.forEach((item) => {
+      if (item["length"] != 0) {
+        item.forEach(async function (element) {
+          const authorName = await whoIsAuthor(element["userId"]);
+          responseDiv.appendChild(
+            addCard(element["title"], element["body"], authorName)
```
Comment (Mr-MRF-Dev) [inline mohammad/JS/JS-dataFetchingAndDOM/search.js:85]: where is `addCard` ??????????

**Review by Mr-MRF-Dev — CHANGES_REQUESTED** [review]

File: mohammad/JS/JS-dataFetchingAndDOM/add.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    }),
+    headers: {
+      "Content-type": "application/json;; charset=UTF-8",
+    },
+  })
+    .then((res) => res.json())
+    .then((json) => {
+      const postID = json["id"];
+      window.alert(`The post by ${postID} post id was added successfully.`);
+      // this method is used for redirect from one page to another
```
[the remark this answers is not in the document]
Comment (mohammadkazemy) [reply mohammad/JS/JS-dataFetchingAndDOM/add.js:24]: I can't understand what you mean :(

**Review by Mr-MRF-Dev — APPROVED** [review]
:)

File: mohammad/JS/JS-dataFetchingAndDOM/add.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
+const url = "https://jsonplaceholder.typicode.com";
+
+function submit() {
+  const title = document.getElementById("title").value;
+  const body = document.getElementById("body").value;
+  const author = document.getElementById("author").value;
```
Comment (mkermani144) [inline mohammad/JS/JS-dataFetchingAndDOM/add.js:6]: This field seems not to be used.

File: mohammad/JS/JS-dataFetchingAndDOM/search.js [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+            addSearchedDiv(element["title"], element["body"], authorName)
+          );
+        });
+      }
+    });
+  });
+}
+
+//debounce function is something like a loop that call a function in a specific period
+function debounce(func, delay) {
```
Comment (mkermani144) [inline mohammad/JS/JS-dataFetchingAndDOM/search.js:94]: Debouncing wasn't part of your issue and is a sign of being a copycat. I don't care about it btw, it's totally up to you.

**Review by mkermani144 — CHANGES_REQUESTED** [review]

File: mohammad/JS/JS-dataFetchingAndDOM/search.js (hunk 7 above)
Comment (mohammadkazemy) [reply mohammad/JS/JS-dataFetchingAndDOM/search.js:94]: In the search section, we need debouncing to call the apiSearch function that the user may enter a different input every few seconds.

File: mohammad/JS/JS-dataFetchingAndDOM/add.js (hunk 6 above)
Comment (mohammadkazemy) [reply mohammad/JS/JS-dataFetchingAndDOM/add.js]: you are right
at first in my first implementation I need this field but in my last implementation I don't need this field and I forgot to delete it.

**Review by mkermani144 — APPROVED** [review]

**Pull request merged**